NUT-04: Add more details about `expiry` (#245)
cashubtc/nuts: 04.md

}
```

Where `quote` is the quote ID and `request` is the payment request to fulfill. `amount` and `unit` correspond to the same values provided in the request.

`state` is an enum string field with possible values `"UNPAID"`, `"PAID"`, `"ISSUED"`:

- `"UNPAID"` means that the quote's request has not been paid yet.
- `"PAID"` means that the request has been paid.
- `"ISSUED"` means that the quote has already been issued.

> `expiry` is the Unix timestamp until which the `request` can be paid (i.e. the bolt11 invoice expiry). Note that the only purpose of this field is to be informative to the user and can not be enforced by the mint.

Reviewer: ```suggestion
`expiry` is the Unix timestamp until which the `request` can be paid (i.e. the bolt11 invoice expiry). Note that the only purpose of this field is to be informative to the user and **MUST NOT** be enforced by the mint.
```

> `expiry` is the Unix timestamp until which the mint quote is valid. Note that `expiry` is currently not enforced, instead it is more of a helpful hint from the mint, which allows wallets to tell when the invoice (`request`) expires without having to parse it. It generally holds that, if a quote's `request` can be paid, the quote should transition to `PAID` regardless of `expiry`. Similarly, if a mint quote is `PAID`, the wallet should be able to claim the proofs (i.e. by minting the tokens, which transitions the quote to `ISSUED`) regardless of `expiry`. The field could however be enforced in the future, for example in setups where `request` doesn't have a built-in expiry, for example an onchain address.
> 

Reviewer 2: It's a bit too much text for a field without use. I propose:
```suggestion
`expiry` is the Unix timestamp until which the mint quote is valid. Note that the only purpose of this field is to be informative to the user and can not be enforced by the mint.
```
